jupyterkernel/lexer/cadabra.js

```diff
@@ -0,0 +1,147 @@
+// CodeMirror, copyright (c) by Marijn Haverbeke and others
+// Distributed under an MIT license: https://codemirror.net/LICENSE
+
+// Adapted by Fergus Baker for Cadabra 2 (c) Kasper Peeters
+
+(function(mod) {
+  if (typeof exports == "object" && typeof module == "object") // CommonJS
+    mod(require("../../lib/codemirror"));
+  else if (typeof define == "function" && define.amd) // AMD
+    define(["../../lib/codemirror"], mod);
+  else // Plain browser env
+    mod(CodeMirror);
+})(function(CodeMirror) {
+  "use strict";
+
+  // lexing starts here!
+  CodeMirror.defineMode('cadabra', function(cfg, cfg_parser) {
+    // use the python code mirror mode
+    var pythonmode = CodeMirror.getMode(cfg, {name:'python'});
+
+    function isWhitespace(char) {
+      return !(char) || char == ' ' || char == '\n' || char == '\t'
+    }
+
+    function eatWord(stream) {
+      var count = 0;
+      while(stream.eat(/[\w\d]+/g)){
+        // eat word
+        count++;
+      }
+      return count;
+    }
+
+    function cadabraToken(stream, state) {
+      // big long basic if/else parser
+
+      // get current token string
+      var current = stream.current()
+
+      // ignore if string; use lookbehind to see if escaped
+      if (current.match(/(?<!\\)".*(?<!\\)"/g)) {
```

Comment:
This line is [broken on Safari](https://stackoverflow.com/questions/51568821/works-in-chrome-but-breaks-in-safari-invalid-regular-expression-invalid-group). Look-back regex isn't yet fully supported.

A possible fix is 
```js
if (current.match(/\\?".*\\?"/g)) {
```